OSDOCS-3969c Alibaba VPC Installation

GroceryBoyJr opened the pull request:
Status: LGTM received from QE.  I have SME and QE LGTM. Commits are squashed. Peer review and two merge reviews complete. **Final Merge Review Requested**

Problem: This PR adds a new section documenting Alibaba VPC installation

Version(s): 4.10+. Original branch was from "main" before 4.12 freeze.

Issue:
How to install OpenShift into an Alibaba VPC https://issues.redhat.com/browse/OSDOCS-3969

FORMER PR for this issue:
https://github.com/openshift/openshift-docs/pull/44593

Link to docs preview:
https://55223--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_alibaba/installing-alibaba-vpc.html

Additional information:
Gaurav Singh, PM. QE: Jianli Wei, SME: Thomas Wiest (Thomas has left the company but provided LGTM before departure.)

jianli-wei commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1735,55 +1608,104 @@ ifdef::alibabacloud[]
 |`platform.alibabacloud.defaultMachinePlatform.instanceType`
-|For compute machines, the configuration used when installing on Alibaba Cloud.
-|String. For example `ecs.g6.large`.
+|For both compute machines and control plane machines, the ECS instance type used to create the ECS instance. Example: `ecs.g6.large`
```
Suggest to replace `ecs.g6.large` with `ecs.g6.xlarge` for `platform.alibabacloud.defaultMachinePlatform.instanceType`.

GroceryBoyJr replied:
@jianli-wei Done, thank you sir.

jianli-wei commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -34,27 +38,32 @@ ifdef::google-cloud-platform[]
 
 You must have:
 
 * Extracted and prepared the `ccoctl` binary.
```
Suggest to tell a link on how to get `ccoctl`, e.g. https://docs.openshift.com/container-platform/4.12/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html#cco-ccoctl-configuring_cco-mode-gcp-workload-identity.

GroceryBoyJr replied:
@jianli-wei I have added some information on how to acquire the ccoctl binary. Thank you sir.

GroceryBoyJr commented:
Hello @jianli-wei PTAL.

jianli-wei commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -34,27 +38,79 @@ ifdef::google-cloud-platform[]
++
+[source,terminal]
+----
+$ oc image extract $CCO_IMAGE --file="/usr/bin/ccoctl" -a ~/.pull-secret
```
Suggest to insert another two steps before this step, which tell how to get the `$CCO_IMAGE`, thanks! 

FYI
1. Obtain the OpenShift Container Platform release image by running the following command:
......
2. Obtain the CCO container image from the OpenShift Container Platform release image by running the following command:
......
3. Extract the `ccoctl` binary from the CCO container image within the OpenShift Container Platform release image by running the following command:
......
4. Change the permissions to make `ccoctl` executable by running the following command:
......

GroceryBoyJr replied:
This is wonderful, @jianli-wei and I very much appreciate your guidance on this! I have added these steps and used the information you directed me to. PTAL!

GroceryBoyJr commented:
@jianli-wei I plan to check back on the PR before our Monday, since your Monday happens before ours. I'd like to see if we can get some acceleration of the PR going toward merge. Thank you so much!

jianli-wei commented:
/lgtm

GroceryBoyJr commented:
/label peer-review-needed

mjpytlak commented on modules/cluster-entitlements.adoc:
```diff
@@ -117,6 +118,12 @@ endif::[]
 ifeval::["{context}" == "installing-restricted-networks-gcp"]
 :restricted:
 endif::[]
+ifeval::["{context}" == "installing-restricted-networks-alibaba-installer-provisioned"]
```
We do not support installing an Alibaba cluster in a restricted environment. Did someone tell you to add this `ifeval`?

GroceryBoyJr replied:
@mjpytlak This edit was not my edit. Actually looks intentional, doesn't it?

mjpytlak commented on modules/installation-alibaba-config-yaml.adoc:
```diff
@@ -37,26 +38,31 @@ metadata:
   machineNetwork:
   - cidr: 10.0.0.0/16
-  networkType: OVNKubernetes <2>
+  networkType: OpenShiftSDN
```
This is not correct. The default value is `OVNKubernetes`, which was introduced in 4.12 by https://github.com/openshift/openshift-docs/pull/54319

GroceryBoyJr replied:
I agree completely. This file has been refreshed and my changes re-entered.

mjpytlak commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -116,9 +118,6 @@ endif::[]
 ifeval::["{context}" == "installing-gcp-vpc"]
 :gcp:
 endif::[]
-ifeval::["{context}" == "installing-gcp-shared-vpc"]
```
As it stands, you are continuing to delete GCP and IBM Cloud VPC content, plus what appears to be other updates. For example, you seem to be deleting content around bare metal content.

GroceryBoyJr replied:
It's completely ridiculous, 90% of these changes are not mine, including deleting GCP and IBM Cloud VPC. I'll save out my changes and revert this file.

GroceryBoyJr replied:
@mjpytlak this has been fixed

mjpytlak commented on modules/installation-configure-proxy.adoc:
```diff
@@ -62,6 +63,9 @@ ifeval::["{context}" == "installing-aws-china-region"]
 :aws:
 :aws-china:
 endif::[]
+ifeval::["{context}" == "installing-alibaba-vpc"]
```
You have added an `ifeval` for your VPC assembly, but I do not see any specific content that would require the condition.

GroceryBoyJr replied:
The file has been removed from this PR (as expected).

mjpytlak commented on modules/installation-launching-installer.adoc:
```diff
@@ -90,10 +94,6 @@ ifeval::["{context}" == "installing-aws-default"]
 :no-config:
 :aws:
 endif::[]
-ifeval::["{context}" == "installing-aws-localzone"]
```
Removing this `ifeval` will break the AWS Local Zones doc that shipped with 4.12.

GroceryBoyJr replied:
@mjpytlak this file has been replaced with new one and no longer has that problem.

mjpytlak reviewed (commented):
@GroceryBoyJr I think the problem with your local files still persists. As it stands, this PR still would be removing content specific to GCP, IBM Cloud VPC, and AWS Local Zones doc.

jeana-redhat commented:
Dropping my review because Mike has way better context here

/remove-label peer-review-in-progress

jeana-redhat commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -72,22 +154,23 @@ $ oc adm release extract \
 endif::aws-sts[]
 ifdef::google-cloud-platform[]
 ----
+$ RELEASE_IMAGE=$(./openshift-install version | awk '/release image/ {print $3}')
```
This caught my eye while closing tabs - only one command per block, so if you want to set this variable it needs to be added as a step above this one.

GroceryBoyJr replied:
@jeana-redhat Thank you! I've made the change, much appreciated!

jeana-redhat replied:
- Right now, lines 156-158 & 163 are in the GCP content. I believe they belong in the Alibaba content.
- While the `$ RELEASE_IMAGE=$` command now has its own block, it still needs to be an actual distinct step (one command per step). Something like this:
```
. Set the `$RELEASE_IMAGE` variable by running the following command:
+
[source,terminal]
----

----

. Extract the list of `CredentialsRequest` objects from the {product-title} release image by running the following command:
+
[source,terminal]
----
$ oc adm release extract \
--credentials-requests \
--cloud=alibabacloud \
--to=<path_to_directory_with_list_of_credentials_requests>/credrequests \ <1>
--from=$RELEASE_IMAGE
----

GroceryBoyJr commented:
/remove-label peer-review-in-progress
/label peer-review-needed

bscott-rh commented:
/label peer-review-in-progress

bscott-rh commented on _topic_maps/_topic_map.yml:
```diff
@@ -142,6 +142,8 @@ Topics:
       File: installing-alibaba-customizations
     - Name: Installing a cluster on Alibaba Cloud with network customizations
       File: installing-alibaba-network-customizations
+    - Name: Installing a cluster on Alibaba Cloud with VPC customizations
```
```suggestion
    - Name: Installing a cluster on Alibaba Cloud into a shared VPC
```
I'm suggesting this because it's how we named the GCP shared VPC installation assembly. Up to you.

GroceryBoyJr replied:
@bscott-rh Thank you! Done!

bscott-rh commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -34,60 +38,148 @@ ifdef::google-cloud-platform[]
 endif::google-cloud-platform[]
-ifdef::alibabacloud-default,alibabacloud-customizations[]
+ifdef::alibabacloud-default,alibabacloud-customizations,alibabacloud-vpc[]
+
```
```suggestion
```
I think this extra line is messing up the formatting. See the preview: https://55223--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_alibaba/installing-alibaba-vpc.html#using-the-ccoctl-program The code block in step 1 isn't indented or syntax-highlighted. 

I think this may also fix the broken formatting with the successive steps, but please double check in your local preview if this extra line is the issue.

GroceryBoyJr replied:
Agreed, agreed. Thank you for this, it got by me and is :-1:

bscott-rh reviewed (changes requested):
Picked up a couple of formatting issues and some small typos/word replacements.

GroceryBoyJr commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -246,7 +250,7 @@ Required installation configuration parameters are described in the following ta
 
 |`apiVersion`
-|The API version for the `install-config.yaml` content. The current version is `v1`. The installation program may also support older API versions.
+|The API version for the `install-config.yaml` content. The current version is `v1`. The installer may also support older API versions.
```
@bscott-rh thank you, done!

GroceryBoyJr commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -34,60 +38,148 @@ ifdef::google-cloud-platform[]
+
+.Output
+
++
```
@bscott-rh Thank you! Done!

GroceryBoyJr commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -288,7 +380,7 @@ where:
 
 `<path_to_ccoctl_output_dir>`:: Specifies the directory created by the `ccoctl alibabacloud create-ram-users` command.
 `<path_to_installation>dir>`:: Specifies the directory in which the installation program creates files.
```
Wow! EXCELLENT catch!

bscott-rh commented on modules/installation-launching-installer.adoc:
```diff
@@ -54,6 +55,9 @@ ifeval::["{context}" == "installing-alibaba-network-customizations"]
 :custom-config:
 :single-step:
 endif::[]
+ifeval::["{context}" == "installing-alibaba-vpc"]
+:custom-config:
+endif::[]
```
```suggestion
ifeval::["{context}" == "installing-alibaba-vpc"]
:custom-config:
:single-step:
endif::[]
```
Please add `:single-step:` here and `!:single-step:` after line 557. This should correct the formatting bug in the "deploying the cluster" section:

![Screenshot 2023-02-10 at 3 32 47 PM](https://user-images.githubusercontent.com/93731574/218192068-b0c7991e-9eeb-4a2c-a781-3e3e0e7e4428.png)

GroceryBoyJr replied:
@bscott-rh This is done, thanks so much for this tip!

bscott-rh commented:
LGTM!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

GroceryBoyJr commented:
/label merge-review-requested

GroceryBoyJr commented:
/label merge-review-needed

jeana-redhat commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -34,27 +38,104 @@ ifdef::google-cloud-platform[]
+[source,terminal]
+----
+$ chmod 775 ccoctl
+----
+
+.Verification
+
+. To verify that `ccoctl` is ready to use, display the help file by running the following command:
++
+[source,terminal]
+----
+$ ccoctl --help
+----
+
+.Output
+
+[source,terminal]
+----
+OpenShift credentials provisioning tool
+
+Usage:
+  ccoctl [command]
+
+Available Commands:
+  alibabacloud Manage credentials objects for alibaba cloud
+  aws          Manage credentials objects for AWS cloud
+  gcp          Manage credentials objects for Google cloud
+  help         Help about any command
+  ibmcloud     Manage credentials objects for IBM Cloud
+  nutanix      Manage credentials objects for Nutanix
+
+Flags:
+  -h, --help   help for ccoctl
+
+Use "ccoctl [command] --help" for more information about a command.
+----
+
+== Using the `ccoctl` program
+
+
```
This appears to duplicate the existing module `cco-ccoctl-configuring.adoc`. Use that module instead of duplicating it here for two reasons:

1. We don't want to have to maintain diverging versions of this procedure - leverage mod docs here for less headaches :slightly_smiling_face: 
2.  By duplicating it here, it is now showing up in places it should not, such as in [Creating AWS resources with a single command](https://55223--docspreview.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#cco-ccoctl-creating-at-once_cco-mode-sts)

This should not be merged until this issue is resolved.

GroceryBoyJr replied:
@jeana-redhat This has been done, you helped me interactively when you came to the tower, thank you so much.

mjpytlak commented:
@GroceryBoyJr Looks like you have addressed all of my feedback. I know you are addressing some feedback from Jeana. Once Jeana approves your updates, I believe you are OK to merge this. I know this has been through the review wringer, so I was looking for anything that might break other assemblies. I did not see anything.

CC: @bscott-rh @jeana-redhat

GroceryBoyJr commented:
> @GroceryBoyJr Looks like you have addressed all of my feedback. I know you are addressing some feedback from Jeana. Once Jeana approves your updates, I believe you are OK to merge this. I know this has been through the review wringer, so I was looking for anything that might break other assemblies. I did not see anything.
> 
> CC: @bscott-rh @jeana-redhat

@mjpytlak Thank you so much!

GroceryBoyJr commented:
/label merge-review-needed

skrthomas commented on modules/installation-custom-alibaba-vpc.adoc:
```diff
@@ -0,0 +1,52 @@
+[id="installation-custom-alibaba-vpc_{context}"]
+= Using a custom VPC
+
+In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you can avoid limit constraints in new accounts more easily adhere to your organization's operational constraints. If you cannot obtain the infrastructure creation permissions that  ...
```
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you can avoid limit constraints in new accounts and more easily adhere to your organization's operational constraints. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```

skrthomas replied:
I think you need to add "and" here

GroceryBoyJr replied:
@skrthomas Thank you, done!

skrthomas reviewed (commented):
One nit :(

skrthomas commented:
/cherrypick enterprise-4.12

jianli-wei commented:
@GroceryBoyJr Sorry, if possible please address my last comment below, thanks in advance! 

The page title is "_Installing a cluster on Alibaba Cloud into an existing VPC_", but the left-side menu shows it as "_Installing a cluster on Alibaba Cloud into a shared VPC_". Suggest to replace "**a shared**" with "**an existing**", because "shared VPC" refers to a bit different concept.
